### PR TITLE
Mobile ios

### PR DESCRIPTION
**Pull request by jchapmanio**
* Renamed Resources/AndroidClient to Resources/MobileClient (both Android and iOS use same prefabs for now)

* Added iOSClientWorker prefab and iOSWorkerConnector script. The script is basically the same as AndroidWorkerConnector with some appropriate changes, the biggest being that the ip address to use is a property on the component rather than using the GDK build configuration.

* Copied link.xml file over from gdk-for-unity

* New scene, FPS-iOSClient

* Some project setting changes as a result of making Unity happy to build for iOS.

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        {
+#if UNITY_IOS
+            if (!string.IsNullOrEmpty(IpAddress))
+            {
+                return IpAddress;
+            }
+
+            if (Application.isMobilePlatform && iOSDeviceInfo.ActiveDeviceType == MobileDeviceType.Virtual)
+            {
+                // TODO What should this return? No "EmulatorDefaultCallbackIp" in iOSDeviceInfo
```
Comment (gkassabli) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: iOS sim Ip is the same as local machine, so we can just return RuntimeConfigDefaults.ReceptionistHost.
We probably don't need this if

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 2]
```diff
@@ -156,6 +156,7 @@ PlayerSettings:
   applicationIdentifier:
     Android: io.improbable.gdk
     Standalone: com.improbable.gdk.starterproject
+    iOS: com.improbable.gdk.jon
```
Comment (gkassabli) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: why not .starterproject?

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs (hunk 1 above)
Comment (firtina-improbable) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: @gkassabli this comment seems truncated

Comment (gkassabli) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: Haha, it wasn't, but ending the comment with if and without period is indeed confusing :)

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 2 above)
Comment (jchapmanio) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: Darnit I thought I'd removed my name already haha. `com.improbable.starterproject`, then?

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 3]
```diff
@@ -154,8 +154,9 @@ PlayerSettings:
   androidSupportedAspectRatio: 1
   androidMaxAspectRatio: 2.1
   applicationIdentifier:
-    Android: io.improbable.gdk
+    Android: io.improbable.gdk.starterproject
     Standalone: com.improbable.gdk.starterproject
+    iOS: com.improbable.gdk.starterproject
```
Comment (paulbalaji) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: why is the Android one `io..` but the Standalone/iOS ones `com...`?

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs [hunk 4]
```diff
@@ -7,6 +7,7 @@
 using Improbable.Gdk.PlayerLifecycle;
 using Improbable.Worker.CInterop;
 using UnityEngine;
+
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: super nit: no need for this new line

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+using Improbable.Gdk.Core;
+using Improbable.Gdk.GameObjectCreation;
+using Improbable.Gdk.GameObjectRepresentation;
+using Improbable.Gdk.Mobile;
+using Improbable.Gdk.PlayerLifecycle;
+using Improbable.Worker.CInterop;
+using UnityEngine;
+#if UNITY_IOS
+using Improbable.Gdk.Mobile.iOS;
+
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: super nit: remove line :D

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        protected virtual void LoadWorld()
+        {
+            var workerSystem = Worker.World.GetExistingManager<WorkerSystem>();
+            var worldSize = workerSystem.Connection.GetWorkerFlag("world_size");
+
+            if (worldSize != Small && worldSize != Large)
+            {
+                workerSystem.LogDispatcher.HandleLog(LogType.Error,
+                    new LogEvent(
+                            "Invalid world_size worker flag. Make sure that it is either small, medium, or large,")
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: should this be "either small or large"? we don't seem to check for medium

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                        .WithField("world_size", worldSize));
+                return;
+            }
+
+            var levelToLoad = worldSize == Large ? LargeLevelPrefab : SmallLevelPrefab;
+
+            if (levelToLoad == null)
+            {
+                Debug.LogError("The level to be instantiated is null.");
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: also use the logdispatcher here. It should still work, but better to stay consistent with how we log :)

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 3 above)
Comment (jessicafalk) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: yep, that should be the same one for both :)

File: workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        }
+
+        public async void Reconnect()
+        {
+            await AttemptConnect();
+        }
+
+        protected async Task AttemptConnect()
+        {
+            await Connect(GetWorkerType(), new ForwardingDispatcher()).ConfigureAwait(false);
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/iOSWorkerConnector.cs]: you could just inline `WorkerUtils.iOSClient` no need to define a new method for it

File: workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs (hunk 4 above)
Comment (jchapmanio) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/AndroidWorkerConnector.cs]: The new line gets added by the code cleanup as its before an `#if`, should I leave it in for that reason?

**Comment by jchapmanio** [comment]
Actioned review suggestions, except one (see comment). Will test on iPhone 7 Monday with Matt!

**Comment by jessicafalk** [comment]
two things that need to be added:
1. add the symbol `UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP` for iOS in the Player Settings > Scripting Define Symbols
2. add the following to the link.xml:
```
<assembly fullname="Improbable.Gdk.Health" preserve="all"/>
<assembly fullname="Improbable.Gdk.Guns" preserve="all"/>
<assembly fullname="Improbable.Gdk.Movement" preserve="all"/>
<assembly fullname="Improbable.Gdk.ObjectPooling" preserve="all"/>
<assembly fullname="Improbable.Gdk.Ragdoll" preserve="all"/>
<assembly fullname="Improbable.Gdk.StandardTypes" preserve="all"/>
```

**Comment by jchapmanio** [comment]
UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP is a shared property among all players, FYI (in case that affects anything)

**Pull request merged**